syncd/syncd.cpp

```diff
@@ -3569,174 +3548,6 @@ void processFlexCounterEvent(
-    std::cout << "        Provide profile map file" << std::endl;
-    std::cout << "    -i --countersInterval interval" << std::endl;
-    std::cout << "        Provide counter thread interval" << std::endl;
-    std::cout << "    -t --startType type" << std::endl;
-    std::cout << "        Specify cold|warm|fast|fastfast start type" << std::endl;
-    std::cout << "    -u --useTempView:" << std::endl;
-    std::cout << "        Use temporary view between init and apply" << std::endl;
-    std::cout << "    -S --disableExitSleep" << std::endl;
-    std::cout << "        Disable sleep when syncd crashes" << std::endl;
-    std::cout << "    -U --eableUnittests" << std::endl;
-    std::cout << "        Metadata enable unittests" << std::endl;
-    std::cout << "    -C --eableConsistencyCheck" << std::endl;
-    std::cout << "        Enable consisteny check DB vs ASIC after comparison logic" << std::endl;
-#ifdef SAITHRIFT
-    std::cout << "    -r --rpcserver"           << std::endl;
-    std::cout << "        Enable rpcserver"      << std::endl;
-    std::cout << "    -m --portmap"             << std::endl;
-    std::cout << "        Specify port map file" << std::endl;
-#endif // SAITHRIFT
-    std::cout << "    -s --syncMode"             << std::endl;
-    std::cout << "        Enable synchronous mode" << std::endl;
-    std::cout << "    -h --help" << std::endl;
-    std::cout << "        Print out this message" << std::endl;
-}
-
-void handleCmdLine(int argc, char **argv)
-{
-    SWSS_LOG_ENTER();
-
-    options.disableExitSleep = false;
-    options.enableUnittests = false;
-    options.syncMode = false;
-
-#ifdef SAITHRIFT
-    options.run_rpc_server = false;
-    const char* const optstring = "dNUCt:p:i:rm:huSs";
-#else
-    const char* const optstring = "dNUCt:p:i:huSs";
-#endif // SAITHRIFT
-
-    while(true)
-    {
-        static struct option long_options[] =
-        {
-            { "useTempView",             no_argument,       0, 'u' },
-            { "diag",                    no_argument,       0, 'd' },
-            { "startType",               required_argument, 0, 't' },
-            { "profile",                 required_argument, 0, 'p' },
-            { "help",                    no_argument,       0, 'h' },
-            { "disableExitSleep",        no_argument,       0, 'S' },
-            { "enableUnittests",         no_argument,       0, 'U' },
-            { "enableConsistencyCheck",  no_argument,       0, 'C' },
-#ifdef SAITHRIFT
-            { "rpcserver",               no_argument,       0, 'r' },
-            { "portmap",                 required_argument, 0, 'm' },
-#endif // SAITHRIFT
-            { "syncMode",                no_argument,       0, 's' },
-            { 0,                         0,                 0,  0  }
-        };
-
-        int option_index = 0;
-
-        int c = getopt_long(argc, argv, optstring, long_options, &option_index);
-
-        if (c == -1)
```

Comment:
>gPortMap [](start = 8, length = 8)

I search the code and no one is using gPortMap. Can you remove it at all?

Reply:
no i can't, gPortMap is used in sai thirft rpc server and it;s linked agains this object here, i sent email to Pavel to maybe figure out some other to make this like call function to set actual port map or set parh to it to saithirft server, also there is similar object gSwitchId which is also uses in saithirt and it's global object providing functionality for multiple switches, then this approach will not  work, i will forward you email